src/components/Chat/type.d.ts

```diff
@@ -0,0 +1,8 @@
+export interface ChatBoxProps {
+  isOpen: boolean;
+  setIsOpen: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export interface ChatButtonProps {
+  setIsOpen: React.Dispatch<React.SetStateAction<boolean>>;
+}
```

Comment:
```suggestion
type ChatModalStateType = React.Dispatch<React.SetStateAction<boolean>>

export interface ChatBoxProps {
  isOpen: boolean;
  setIsOpen: ChatModalStateType;
}

export interface ChatButtonProps {
  setIsOpen: ChatModalStateType;
}
```
혹은
```suggestion
export interface ChatBaseType {
  isOpen: boolean;
  setIsOpen: React.Dispatch<React.SetStateAction<boolean>>;
}
```
사용하는 부분에서
`Pick<ChatBaseType, setIsOpen>` 을 가져다가 사용해도 괜찮을 것 같습니다.

Reply:
추후에 ButtonProps 에 추가가 될수도 있어서 일단 만들어 두긴 했는데 없어도 될 까요?

Reply:
나중에 값이 많이 추가된다면, 그냥 그대로 두셔도 괜찮을 것 같습니다.